src/transforms.ts

```diff
@@ -0,0 +1,60 @@
+import type { LatexNode } from './types.js';
+
+function ousetMutate(content: LatexNode[]) {
+  return content
+    .map((node, index) => {
+      if (
+        node.type === 'macro' &&
+        typeof node.content === 'string' &&
+        ['overset', 'underset'].includes(node.content)
+      ) {
+        const next = content?.[index + 1];
+        const nextNext = content?.[index + 2];
+        if (
+          next &&
+          typeof next !== 'string' &&
+          next.type === 'group' &&
+          next.content &&
+          nextNext &&
+          typeof nextNext !== 'string' &&
+          nextNext.type === 'group' &&
+          nextNext.content
+        ) {
+          node.args = [
+            {
+              type: 'argument',
+              content: nextNext.content,
+              openMark: '{',
+              closeMark: '}',
+            },
+            {
+              type: 'argument',
+              content: next.content,
+              openMark: '{',
+              closeMark: '}',
+            },
+          ];
+          next.type = '__delete__';
+          nextNext.type = '__delete__';
+        }
+      } else if (node.type !== '__delete__') {
+        return ousetTransform(node);
+      }
+      return node;
+    })
+    .filter((node) => node?.type !== '__delete__');
+}
+
+function ousetTransform(tree: LatexNode) {
```

Comment:
We could maybe do this in the parse/walk step instead of introducing another transform step... But for me, this was more understandable...